base/src/types.rs

```diff
@@ -229,6 +217,7 @@ pub enum BuiltinType {
     Array,
     /// Type constructor for functions, `(->) a b : Type -> Type -> Type`
     Function,
+    Type,
```

Comment:
I must admit I am a bit out of my depth when it comes to this stuff. That being said, as far as I understand does not universes come into play when there may be mixing of types of multiple universes. As long as types and kinds (and values) are separate as they are now universes should be implicitly defined simply from the context the type are in. Might be wrong though :worried: